### PR TITLE
feat: Update Boilerplate to match Refactored User Stories

### PR DESCRIPTION
**Pull request by raisedadead**
This recreates https://github.com/freeCodeCamp/boilerplate-project-issuetracker/pull/13

From the OG PR:

> @nhcarrigan and I refactored the tests and user stories for this project as described here:
> https://github.com/freeCodeCamp/freeCodeCamp/pull/39627
> 
> Additionally, this refactor:
> - Removes the User Stories from `index.html`
> - Adds the `.gitignore` and `sample.env` files
> - Updates the favicon
> - Removes lingering references to Hyperdev
> - Removes `zombie` and `helmet` packages as they are no longer used
> 
> Ref: freeCodeCamp/freeCodeCamp#39721

### CODE REVIEW

**Comment by naomi-lgbt** [comment]
I believe this should not be merged until [#39627](https://github.com/freeCodeCamp/freeCodeCamp/pull/39627) hits production.

File: assertion-analyser.js
Comment (ShaunSHamilton) [inline assertion-analyser.js]:
```suggestion
  while(obj == objParser(str)) {
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
I was looking at my own PR, and noticed this, because the linter did something weird. But, is this not a problem?

@moT01  and @RandellDawson I have tested my project, as well as most of the others, and have seen no breaks, but this does not look correct. Am i missing something?

**Comment by SaintPeter** [comment]
@Sky020 I think that `while(obj =` bug is probably in every single project that uses the `assertion-analyzer.js`.

File: assertion-analyser.js
Comment (ShaunSHamilton) [inline assertion-analyser.js]:
```suggestion
  while(obj = objParser(str)) {
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
I know I brought this up, but, after testing, I realise it is necessary for the tests to run properly.

File: assertion-analyser.js
Comment (raisedadead) [reply assertion-analyser.js]: That looks like something that can be easily mistaken by someone down the lane as an incorrect conditional without context. 

Can this be made any simpler (even if it's not smart or is more code)?

Comment (ShaunSHamilton) [reply assertion-analyser.js]: Seeing as this is in many of the boilerplates, I would recommend we open an issue about this, first. So far, I just added a comment: https://github.com/freeCodeCamp/boilerplate-mochachai/pull/27/files#diff-c8580e1410209df94bd3f9675480f46dad5973282f1ef427f4cf85ef0a71580bR39

Comment (RandellDawson) [reply assertion-analyser.js]: @raisedadead This is only for maintainers.  The student does not change this file, so is it really necessary to change the working code?  Maybe we can add a comment above the line explaining the purpose of the line?

On a side note, it would be great if we could turn this into a package since it is used in several boilerplate repos.  Everytime we change this code, we have to remember to fix it in all of the repos that use it and since the repos do not have tests, we do not know something is wrong until a student points out something seems to be wrong with the boilerplate.

Comment (ojeytonwilliams) [reply assertion-analyser.js]: Why not:

```js
let obj = objParser(str);
while(obj) {
  data[cnt] = obj.obj;
  str = str.substring(0, obj.start) + '__#' + cnt++ + str.substring(obj.end+1);
  obj = objParser(str);
}
```

It's not the prettiest code, but it avoids the `=` vs `==` confusion.

Assuming we want to change things, I haven't been following this close enough to know.

Comment (SaintPeter) [reply assertion-analyser.js]: @ojeytonwilliams If you submit that as a suggestion I can commit it.  Have you tested locally to make sure it works?

File: assertion-analyser.js
Comment (ojeytonwilliams) [inline assertion-analyser.js]:
```suggestion
  let cnt = 0;
  let data = [];
  let obj = objParser(str);
  while(obj) {
    data[cnt] = obj.obj;
    str = str.substring(0, obj.start) + '__#' + cnt++ + str.substring(obj.end+1);
    objParser(str);
  }
```
I haven't had chance to test it locally, sorry, I will as soon as I can.  For now, here's the suggestion.

File: assertion-analyser.js
Comment (ShaunSHamilton) [reply assertion-analyser.js]: @SaintPeter and @ojeytonwilliams We are likely best not changing anything from what we know works. As Randell mentioned, it would be great to get an issue going to have these tester files be turned into a package, and avoid diverging all of these boilerplates.

Comment (raisedadead) [reply assertion-analyser.js]: > This is only for maintainers. The student does not change this file, so is it really necessary to change the working code?

I do not have context of it, and that's the point. From a cursory look it looks like a missed / incorrect conditional statement. I understand it's just a smart way, but a few months down the lane it pretty easy to miss and someone can make the change for semantic reasons.

Often its best to make it foolproof (trading for more code).

Comment (RandellDawson) [reply assertion-analyser.js]: @nhcarrigan How many boilerplates use this code, because someone should create an issue to track all them where we need to do the same thing?

Comment (naomi-lgbt) [reply assertion-analyser.js]: I will get an issue put together 🙂

Comment (raisedadead) [reply assertion-analyser.js]: If this is a common utility, we are happy to have this as a package. Let me know if you want to go that direction (not necessarily in this PR) and I will set things up. That said I would still recommend keeping it simple like Oliver mentioned in above. 

And more importantly I am totally fine not holding this PR for the nit-pick. It can totally be taken up in its own issue.

Comment (SaintPeter) [reply assertion-analyser.js]: Should I drop that commit or is it fine as is?

Comment (RandellDawson) [reply assertion-analyser.js]: I think we can leave it.  The change works and is documented in issue #40180.  I am going to try and review this PR today along with the corresponding `/learn` PR #39627 today or this weekend.

File: assertion-analyser.js
Comment (naomi-lgbt) [inline assertion-analyser.js]:
```suggestion
    obj = objParser(str);
```

After a call with @RandellDawson, we discovered that the test calls here will not run because the `obj` variable is not reassigned.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: assertion-analyser.js
Comment (RandellDawson) [reply assertion-analyser.js]: @raisedadead @SaintPeter Can you commit this change?  It fixes a bug. We are unable to push any commits to this branch.

Comment (raisedadead) [reply assertion-analyser.js]: Only Rex is able to make changes to this branch. Alternatively, we can land this on a temp branch and make further changes. Let me know.

**Comment by SaintPeter** [comment]
Should be good to go.

**Review by RandellDawson — DISMISSED** [review]
I can now say this looks good to me.

@nhcarrigan @Sky020 If you are ok with the latest changes, go ahead and approve.  We will still need to wait till the `/learn` PR hits production before merging.

File: routes/api.js
Comment (ShaunSHamilton) [inline routes/api.js]: Are we not wanting to remove the inclusion of `mongodb` and all DB-related environment variables?

Comment (RandellDawson) [reply routes/api.js]: @Sky020 Oh yes.  I forgot about that.  Actually, the following can be removed:

```js
const expect = require('chai').expect;
const MongoClient = require('mongodb');
const ObjectId = require('mongodb').ObjectID;

const CONNECTION_STRING = process.env.DB; //MongoClient.connect(CONNECTION_STRING, function(err, db) {});
```
and the following files can be deleted:

- `tests/1_unit-tests.js`
- `.gitconfig`
- `.bashrc`
- `.profile`

@SaintPeter Can you take care of these?

**Comment by SaintPeter** [comment]
I have made the requested changes.

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Comment by RandellDawson** [comment]
The /learn changes have hit production, so I am merging this now.

**Pull request merged**
